src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -4467,13 +4602,12 @@ class CgroupUtils(object):
         """
         pbs.logmsg(pbs.EVENT_DEBUG4, 'Adding jobid %s to cgroup_jobs' % jobid)
         try:
-            with open(self.cgroup_jobs_file, 'r+') as f:
-                joblist = f.readline().split()
-                jobset = set(joblist)
-                jobset.add(jobid)
-                f.seek(0)
-                f.write(' '.join(jobset))
-                f.truncate()
+            with open(self.cgroup_jobs_file, 'r+') as fd:
+                jobdict = eval(fd.read())
```

Comment:
I'm not the biggest fan of `eval()`, have you thought of using `ast.literal_eval`? See https://docs.python.org/2.7/library/ast.html#ast.literal_eval

Reply:
The description of ast.literal_eval() mentions untrusted sources. In this case, the hook itself is not an untrusted source since it populated the file. I'd prefer to leave it as-is.